### PR TITLE
Spotduration

### PR DESCRIPTION
**Pull request by sauvainr**
Allow to passe the spot instance duration.
This to prevent un-wanted build instance to remain alive long than expected.

This would also prevent build instance not being terminated (Situation we encountered in our company).

### CODE REVIEW

File: builder/amazon/common/run_config.go [hunk 1]
```diff
@@ -113,6 +114,11 @@ func (c *RunConfig) Prepare(ctx *interpolate.Context) []error {
 		errs = append(errs, fmt.Errorf("An instance_type must be specified"))
 	}
 
+	if c.BlockDurationMinutes != nil && c.BlockDurationMinutes % 60 != 0 {
```
Comment (azr) [inline builder/amazon/common/run_config.go]: 🌵  Ah, careful c.BlockDurationMinutes is an int64 and cannot be nil.

You can run `make test` to verify.

**Comment by azr** [comment]
Here's a link to the [cli-docs](https://docs.aws.amazon.com/cli/latest/reference/ec2/request-spot-instances.html) as well.

I think adding that `You can't specify an Availability Zone group or a launch group if you specify a duration` would be a good idea too.

File: builder/amazon/common/run_config.go (hunk 1 above)
Comment (sauvainr) [reply builder/amazon/common/run_config.go]: Good point.

Comment (sauvainr) [reply builder/amazon/common/run_config.go]: As a note, it is not handy to validate changes as current master is failing several tests.

**Comment by sauvainr** [comment]
Thanks for the quick feedback @azr

**Comment by azr** [comment]
Hey @sauvainr, sure 👍tried it and it worked !

One last change, look at the travis-ci build; it's complaining that `./builder/amazon/common/run_config.go` is not correctly formatted.

`make fmt` will fix that for you

**Comment by azr** [comment]
thank you @sauvainr 🙂

**Comment by sauvainr** [comment]
No problem, thanks for your help. We will use this new tag as soon as it get in next release.

**Pull request merged**